sdk/src/store.rs

```diff
@@ -1386,26 +1401,76 @@ impl Store {
 
         Ok(hashes)
     }
+
+    // move or copy data from source to dest
+    #[cfg(feature = "file_io")]
+    fn move_or_copy(source: &Path, dest: &Path) -> Result<()> {
+        // copy temp file to asset
+        std::fs::rename(source, dest)
+            // if rename fails, try to copy in case we are on different volumes or output does not exist
+            .or_else(|_| std::fs::copy(source, dest).and(Ok(())))
+            .map_err(Error::IoError)
+    }
+
+    // copy output and possibly the external manifest to final destination
+    #[cfg(feature = "file_io")]
+    fn copy_c2pa_to_output(source: &Path, dest: &Path, remote_type: RemoteManifest) -> Result<()> {
+        match remote_type {
+            crate::claim::RemoteManifest::NoRemote => Store::move_or_copy(source, dest)?,
+            crate::claim::RemoteManifest::SideCar | crate::claim::RemoteManifest::Remote(_) => {
+                // make correct path names
+                let source_asset = source;
+                let source_cai = source_asset.with_extension(MANIFEST_STORE);
+                let dest_cai = dest.with_extension(MANIFEST_STORE);
+
+                Store::move_or_copy(&source_cai, &dest_cai)?; // copy manifest
+                Store::move_or_copy(source_asset, dest)?; // copy asset
+            }
+        }
+        Ok(())
+    }
+
     /// Embed the claims store as jumbf into an asset. Updates XMP with provenance record.
     #[cfg(feature = "file_io")]
     pub fn save_to_asset(
         &mut self,
         asset_path: &Path,
         signer: &dyn Signer,
-        output_path: &Path,
-    ) -> Result<()> {
-        let jumbf_bytes = self.start_save(asset_path, output_path, signer.reserve_size())?;
+        dest_path: &Path,
+    ) -> Result<Vec<u8>> {
+        // set up temp dir, contents auto deleted
+        let td = tempfile::TempDir::new()?;
```

Comment:
FYI, I remember someone reporting that sometimes TempDir doesn't get deleted so we should keep an eye on that.

Reply:
If we had some way of knowing when we can do manifest updates in memory vs in the file, we could avoid the need to use temporary storage here. Something to consider for the future.

Reply:
@gpeacock the primary concern that I'm aware of is in the event of a panic (which is common in test code; hopefully less so otherwise), stack unwinding is not 100% predictable, so a temp-dir might not get cleaned up.

That said, I haven't seen that report, so maybe there's some other failure mode I'm not familiar with.

Reply:
Read the temp dir docs.  They do all they can to address this.  The temp dir changes are in response to a request from Gavin to not corrupt the destination file.  To do that we have to write somewhere else, i.e. a temp location.

Reply:
I don't know that there is anything we need to do at this point, I was just noting that I had this code in claim_tool and the issue was brought up by AdobeStock. It may have been under test conditions.